File: codewars/Duplicate Encoder/duplicate_encoder.py
def duplicate_encode(s):
    s = s.lower()
    
    char_count = {}
    for char in s:
        if char in char_count:
            char_count[char] += 1
        else:
            char_count[char] = 1
    
    result = ""
    for char in s:
        if char_count[char] > 1:
            result += ")"
        else:
            result += "("
    
    return result

File: codewars/Duplicate Encoder/Duplicate_Encoder.ts
export function duplicateEncode(word: string): string {
    const lowerCaseWord = word.toLowerCase();
    const charCount: { [key: string]: number } = {};
    for (const char of lowerCaseWord) {
        charCount[char] = (charCount[char] || 0) + 1;
    }
    let result = '';
    for (const char of lowerCaseWord) {
        result += charCount[char] === 1 ? '(' : ')';
    }
    return result;
}